### PR TITLE
phpExtensions.intl: icu64 -> icu73

### PR DESCRIPTION
**Pull request by soebbing**
## Description of changes
Updated ICU dependency for PHP packages from icu64 (from 2019) to icu73 (from April 2023).
Changes: https://github.com/unicode-org/icu/compare/release-64-1...release-73-1

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [X] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by drupol — CHANGES_REQUESTED** [review]
Hi !

Thanks for your contribution, much appreciated!

Could you please make sure your commit message match exactly this PR title?

Also, this PR should only contain 1 single commit.

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.intl php82Extensions.intl php83Extensions.intl

File: pkgs/top-level/php-packages.nix [hunk 1]
```diff
@@ -420,7 +420,7 @@ lib.makeScope pkgs.newScope (self: with self; {
         }
         {
           name = "intl";
-          buildInputs = [ icu64 ];
+          buildInputs = [ icu73 ];
```
Comment (shyim) [inline pkgs/top-level/php-packages.nix:423]: What about switching to just `icu`. So it would be always up to date. wdym @drupol

Comment (nyabinary) [reply pkgs/top-level/php-packages.nix:423]: Would this cause any breakage?

Comment (drupol) [reply pkgs/top-level/php-packages.nix:423]: I don't know anything about this thing sorry...

Comment (shyim) [reply pkgs/top-level/php-packages.nix:423]: let's keep it so then :)

**Review by shyim — APPROVED** [review]

**Comment by lelgenio** [comment]
Result of `nixpkgs-review pr 276054` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>arcanist</li>
    <li>drush</li>
    <li>php81Packages.phpcbf</li>
    <li>php81Packages.phpcs</li>
    <li>php82Packages.phpcbf</li>
    <li>php82Packages.phpcs</li>
    <li>php83Packages.phpcbf</li>
    <li>php83Packages.phpcs</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.php.withExtensions-enables-previously-disabled-extensions</li>
  </ul>
</details>
<details>
  <summary>90 packages built:</summary>
  <ul>
    <li>adminer</li>
    <li>php (apacheHttpdPackages.php ,php82)</li>
    <li>bloom</li>
    <li>bookstack</li>
    <li>composer-require-checker</li>
    <li>easyeffects</li>
    <li>easyeffects.debug</li>
    <li>frankenphp</li>
    <li>freshrss</li>
    <li>gnomeExtensions.easyeffects-preset-selector</li>
    <li>icingaweb2</li>
    <li>librenms</li>
    <li>libsForQt5.kcachegrind (plasma5Packages.kcachegrind)</li>
    <li>lsp-plugins</li>
    <li>lsp-plugins.dev</li>
    <li>lsp-plugins.doc</li>
    <li>matomo</li>
    <li>matomo-beta</li>
    <li>n98-magerun</li>
    <li>n98-magerun2</li>
    <li>nagios</li>
    <li>nextcloud-news-updater</li>
    <li>nextcloud-news-updater.dist</li>
    <li>nominatim</li>
    <li>pdepend</li>
    <li>phel</li>
    <li>phoronix-test-suite</li>
    <li>php81</li>
    <li>php81Extensions.intl</li>
    <li>php81Extensions.intl.dev</li>
    <li>php81Packages.box (php82Packages.box ,php83Packages.box)</li>
    <li>php81Packages.castor</li>
    <li>php81Packages.composer</li>
    <li>php81Packages.deployer</li>
    <li>php81Packages.grumphp</li>
    <li>php81Packages.phan</li>
    <li>php81Packages.phing</li>
    <li>php81Packages.phive</li>
    <li>php81Packages.php-cs-fixer</li>
    <li>php81Packages.php-parallel-lint</li>
    <li>php81Packages.phpmd</li>
    <li>php81Packages.phpstan</li>
    <li>php81Packages.psalm</li>
    <li>php81Packages.psysh</li>
    <li>php82Extensions.intl</li>
    <li>php82Extensions.intl.dev</li>
    <li>php82Packages.castor</li>
    <li>php82Packages.composer</li>
    <li>php82Packages.deployer</li>
    <li>php82Packages.grumphp</li>
    <li>php82Packages.phan</li>
    <li>php82Packages.phing</li>
    <li>php82Packages.phive</li>
    <li>php82Packages.php-cs-fixer</li>
    <li>php82Packages.php-parallel-lint</li>
    <li>php82Packages.phpmd</li>
    <li>php82Packages.phpstan</li>
    <li>php82Packages.psalm</li>
    <li>php82Packages.psysh</li>
    <li>php83</li>
    <li>php83Extensions.intl</li>
    <li>php83Extensions.intl.dev</li>
    <li>php83Packages.castor</li>
    <li>php83Packages.composer</li>
    <li>php83Packages.deployer</li>
    <li>php83Packages.grumphp</li>
    <li>php83Packages.phan</li>
    <li>php83Packages.phing</li>
    <li>php83Packages.phive</li>
    <li>php83Packages.php-cs-fixer</li>
    <li>php83Packages.php-parallel-lint</li>
    <li>php83Packages.phpmd</li>
    <li>php83Packages.phpstan</li>
    <li>php83Packages.psalm</li>
    <li>php83Packages.psysh</li>
    <li>phpactor</li>
    <li>phpdocumentor</li>
    <li>phpunit</li>
    <li>pixelfed</li>
    <li>platformsh</li>
    <li>pulseeffects-legacy</li>
    <li>qcachegrind</li>
    <li>robo</li>
    <li>signaturepdf</li>
    <li>snipe-it</li>
    <li>unit</li>
    <li>vimPlugins.phpactor</li>
    <li>wp-cli</li>
    <li>yle-dl</li>
    <li>yle-dl.dist</li>
  </ul>
</details>

**Comment by drupol** [comment]
Why are all these packages broken?

```
drush
php81Packages.phpcbf
php81Packages.phpcs
php82Packages.phpcbf
php82Packages.phpcs
php83Packages.phpcbf
php83Packages.phpcs
```

?

**Comment by lelgenio** [comment]
`phpcbf` and `phpcs` have been replaced by `php-codesniffer`: https://github.com/NixOS/nixpkgs/blob/a426306755afc17c420612523dfce9fa4190a977/pkgs/top-level/php-packages.nix#L225-L228
`drush` can no longer be used as a standalone tool: https://github.com/NixOS/nixpkgs/blob/a426306755afc17c420612523dfce9fa4190a977/pkgs/top-level/aliases.nix#L287-L289

**Comment by drupol** [comment]
Oh you're right, this needs a rebase I guess.

**Comment by soebbing** [comment]
Rebase done!

**Comment by lelgenio** [comment]
Result of `nixpkgs-review pr 276054` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.php.withExtensions-enables-previously-disabled-extensions</li>
  </ul>
</details>
<details>
  <summary>114 packages built:</summary>
  <ul>
    <li>adminer</li>
    <li>adminerevo</li>
    <li>php (apacheHttpdPackages.php ,php82)</li>
    <li>bloom</li>
    <li>bookstack</li>
    <li>composer-require-checker</li>
    <li>davis</li>
    <li>easyeffects</li>
    <li>easyeffects.debug</li>
    <li>engelsystem</li>
    <li>firefly-iii</li>
    <li>flarum</li>
    <li>frankenphp</li>
    <li>freshrss</li>
    <li>gnomeExtensions.easyeffects-preset-selector</li>
    <li>grocy</li>
    <li>icingaweb2</li>
    <li>laravel</li>
    <li>librenms</li>
    <li>libsForQt5.kcachegrind (plasma5Packages.kcachegrind)</li>
    <li>lsp-plugins</li>
    <li>lsp-plugins.dev</li>
    <li>lsp-plugins.doc</li>
    <li>matomo</li>
    <li>matomo-beta</li>
    <li>matomo_5</li>
    <li>movim</li>
    <li>n98-magerun</li>
    <li>n98-magerun2</li>
    <li>nagios</li>
    <li>nextcloud-news-updater</li>
    <li>nextcloud-news-updater.dist</li>
    <li>nominatim</li>
    <li>paratest</li>
    <li>pdepend</li>
    <li>pest</li>
    <li>phel</li>
    <li>phoronix-test-suite</li>
    <li>php81</li>
    <li>php81Extensions.intl</li>
    <li>php81Extensions.intl.dev</li>
    <li>php81Packages.box (php82Packages.box ,php83Packages.box)</li>
    <li>php81Packages.castor</li>
    <li>php81Packages.composer</li>
    <li>php81Packages.composer-local-repo-plugin</li>
    <li>php81Packages.cyclonedx-php-composer</li>
    <li>php81Packages.deployer</li>
    <li>php81Packages.grumphp</li>
    <li>php81Packages.phan</li>
    <li>php81Packages.phing</li>
    <li>php81Packages.phive</li>
    <li>php81Packages.php-codesniffer</li>
    <li>php81Packages.php-cs-fixer</li>
    <li>php81Packages.php-parallel-lint</li>
    <li>php81Packages.phpinsights</li>
    <li>php81Packages.phpmd</li>
    <li>php81Packages.phpstan</li>
    <li>php81Packages.psalm</li>
    <li>php81Packages.psysh</li>
    <li>php82Extensions.intl</li>
    <li>php82Extensions.intl.dev</li>
    <li>php82Packages.castor</li>
    <li>php82Packages.composer</li>
    <li>php82Packages.composer-local-repo-plugin</li>
    <li>php82Packages.cyclonedx-php-composer</li>
    <li>php82Packages.deployer</li>
    <li>php82Packages.grumphp</li>
    <li>php82Packages.phan</li>
    <li>php82Packages.phing</li>
    <li>php82Packages.phive</li>
    <li>php82Packages.php-codesniffer</li>
    <li>php82Packages.php-cs-fixer</li>
    <li>php82Packages.php-parallel-lint</li>
    <li>php82Packages.phpinsights</li>
    <li>php82Packages.phpmd</li>
    <li>php82Packages.phpstan</li>
    <li>php82Packages.psalm</li>
    <li>php82Packages.psysh</li>
    <li>php83</li>
    <li>php83Extensions.intl</li>
    <li>php83Extensions.intl.dev</li>
    <li>php83Packages.castor</li>
    <li>php83Packages.composer</li>
    <li>php83Packages.composer-local-repo-plugin</li>
    <li>php83Packages.cyclonedx-php-composer</li>
    <li>php83Packages.deployer</li>
    <li>php83Packages.grumphp</li>
    <li>php83Packages.phan</li>
    <li>php83Packages.phing</li>
    <li>php83Packages.phive</li>
    <li>php83Packages.php-codesniffer</li>
    <li>php83Packages.php-cs-fixer</li>
    <li>php83Packages.php-parallel-lint</li>
    <li>php83Packages.phpinsights</li>
    <li>php83Packages.phpmd</li>
    <li>php83Packages.phpstan</li>
    <li>php83Packages.psalm</li>
    <li>php83Packages.psysh</li>
    <li>phpactor</li>
    <li>phpdocumentor</li>
    <li>phpunit</li>
    <li>pixelfed</li>
    <li>pulseeffects-legacy</li>
    <li>qcachegrind</li>
    <li>roave-backward-compatibility-check</li>
    <li>robo</li>
    <li>signaturepdf</li>
    <li>simplesamlphp</li>
    <li>snipe-it</li>
    <li>unit</li>
    <li>vimPlugins.phpactor</li>
    <li>wp-cli</li>
    <li>yle-dl</li>
    <li>yle-dl.dist</li>
  </ul>
</details>

**Comment by drupol** [comment]
@ofborg build php81Extensions.intl php82Extensions.intl php83Extensions.intl

**Review by drupol — APPROVED** [review]

**Pull request merged**